ice-rest-catalog/src/main/java/com/altinity/ice/rest/catalog/internal/maintenance/OrphanCleanup.java

```diff
@@ -80,13 +91,21 @@ public void perform(Table table) throws IOException {
                       try {
                         logger.info("Deleting {}", file);
                         tableIO.deleteFile(file);
+                        deletedCount.incrementAndGet();
                         return file;
                       } catch (Exception e) {
                         logger.warn("Failed to delete file {}", file, e);
+                        failedCount.incrementAndGet();
                         return null;
                       }
                     }));
       }
+
+      // Record deletion metrics
+      metrics.recordOrphanFilesDeleted(tableName, deletedCount.get());
+      for (int i = 0; i < failedCount.get(); i++) {
+        metrics.recordOrphanDeleteFailure(tableName);
```

Comment:
any reason it's not `metrics.recordOrphanDeleteFailure(tableName, failedCount.get())`?